tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -823,10 +823,33 @@ def complex_agg(x, agg):
     res = [complex_agg(fft[config["coeff"]], config["attr"]) if config["coeff"] < len(fft)
            else np.NaN for config in param]
     index = ['coeff_{}__attr_"{}"'.format(config["coeff"], config["attr"]) for config in param]
+
+    # Spectral Centroid:
+    fft_abs = np.abs(fft)
+    res += [fft_abs[:len(x)//2].dot(np.arange(0, len(x)//2, 1)) / fft_abs[:len(x)//2].sum()]
+    index += ['centroid']
+
     return zip(index, res)
 
 
 @set_property("fctype", "simple")
+def spectral_centroid(x):
+    """
+    Returns the spectral centroid, which is the average frequency in the absolute fourier transform spectrum.
```

Comment:
can you add a reference to a paper where this feature is used or explained?

Reply:
https://en.wikipedia.org/wiki/Spectral_centroid

Reply:
Thanks! 👍

Reply:
cc @nikhaas